src/main/java/ca/corefacility/bioinformatics/irida/web/controller/api/json/DateJson.java

```diff
@@ -0,0 +1,62 @@
+package ca.corefacility.bioinformatics.irida.web.controller.api.json;
+
+import java.io.IOException;
+import java.sql.Date;
```

Comment:
Why the `java.sql.Date` instead of `java.util.Date`?  I know `java.sql.Date` is a subclass of `util`, but since we use `util` everywhere else why not here?  I remember you mentioning something about the resolution of `sql` vs `util` but that should be handled by the `@Temporal(TemporalType.DATE)` on `Sample` when saving to the database.

Reply:
The use of `sql` vs `util` is handled by `@Temporal` when saving to the database, but not when serializing/de-serializing to JSON. In that case, it's defaulting to serializing/de-serializing as a `java.util.Date`, which actually has different behaviours depending on the version of the package (the version of jackson we have now is serializing as **YYYY-MM-DD**, but the newer version is serializing as a long (milliseconds since epoch)). But, someplace in the serializion process there is an issue with handling time zones, leading to it trying to convert the time of `java.sql.Date` (which is in UTC) to Winnipeg time.

I believe in theory I could deserializing to `java.util.Date` in this class (since I'm implementing the process myself), but I haven't tested to make sure this works.

My reason for de-serializing to `java.sql.Date` is that, when debugging, it makes it explicit the resolution we are storing in this object. That is `java.sql.Date` only stores down to the day (internally, it stores as milliseconds since epoch in UTC time, but sets hour/minute/seconds to 12:00 AM). I also wasn't sure whether or not if I returned `java.util.Date` from the de-serializing process, some other code someplace would then try to convert this value to the proper time zone depending on the locale and so mess up the value.